atlasdb-tests-shared/src/main/java/com/palantir/timelock/paxos/InMemoryTimelockServices.java

```diff
@@ -0,0 +1,152 @@
+import com.palantir.atlasdb.timelock.AsyncTimelockService;
+import com.palantir.atlasdb.timelock.TimeLockServices;
+import com.palantir.atlasdb.util.MetricsManagers;
+import com.palantir.conjure.java.api.config.service.PartialServiceConfiguration;
+import com.palantir.conjure.java.api.config.service.UserAgent;
+import com.palantir.conjure.java.serialization.ObjectMappers;
+import com.palantir.lock.LockService;
+import com.palantir.refreshable.Refreshable;
+import com.palantir.sls.versions.OrderableSlsVersion;
+import com.palantir.timelock.config.ClusterInstallConfiguration;
+import com.palantir.timelock.config.ImmutableClusterInstallConfiguration;
+import com.palantir.timelock.config.ImmutableDefaultClusterConfiguration;
+import com.palantir.timelock.config.ImmutableTimeLockRuntimeConfiguration;
+import com.palantir.timelock.config.PaxosInstallConfiguration;
+import com.palantir.timelock.config.PaxosInstallConfiguration.PaxosLeaderMode;
+import com.palantir.timelock.config.TimeLockInstallConfiguration;
+import com.palantir.timelock.config.TimeLockRuntimeConfiguration;
+import com.palantir.timestamp.TimestampManagementService;
+import com.palantir.timestamp.TimestampService;
+import com.palantir.tritium.metrics.registry.SharedTaggedMetricRegistries;
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+import org.awaitility.Awaitility;
+import org.junit.rules.TemporaryFolder;
+
+public final class InMemoryTimelockServices implements TimeLockServices, Closeable {
+    private static final String USER_AGENT_NAME = "user-agent";
+    private static final String USER_AGENT_VERSION = "3.1415926.5358979";
+    private static final UserAgent USER_AGENT = UserAgent.of(UserAgent.Agent.of(USER_AGENT_NAME, USER_AGENT_VERSION));
+    private static final int THREAD_POOL_SIZE = 128;
+    private static final int BLOCKING_TIMEOUT_MS = 60 * 800; // 0.8 mins to ms
+
+    private final TimeLockServices delegate;
+    private final TimeLockAgent timeLockAgent;
+
+    private InMemoryTimelockServices(TimeLockServices delegate, TimeLockAgent timeLockAgent) {
+        this.delegate = delegate;
+        this.timeLockAgent = timeLockAgent;
+    }
+
+    @Override
+    public void close() {
+        timeLockAgent.shutdown();
+    }
+
+    public static InMemoryTimelockServices create(TemporaryFolder tempFolder) {
+        try {
+            File dataDirectory = tempFolder.newFolder();
+            return create(dataDirectory);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static InMemoryTimelockServices create(File dataDirectory) {
+        PaxosInstallConfiguration paxos = PaxosInstallConfiguration.builder()
+                .dataDirectory(dataDirectory)
+                .leaderMode(PaxosLeaderMode.SINGLE_LEADER)
+                .isNewService(false)
```

Comment:
Interesting that this works with `isNewService` set to false; I'd have thought you need it to be true for it to create the paxos directories

Reply:
it seemed not to work with `isNewService = true` initially - might have been a red herring. Is it important which way we set this flag?

Reply:
This is one of those things where my lack of intimate timelock knowledge means I'm not sure. I was under the impression we needed something along these lines when installing Timelock (and so this would be similar), but maybe not for fresh installs (vs adding a node, say).

Reply:
So the tests fail with isNewService=true, because `tempFolder.newFolder()` creates the data directory. We end up with this failure:
```
com.palantir.logsafe.exceptions.SafeIllegalArgumentException: This timelock server has been configured as a new stack but the Paxos data directories already exist. Almost surely this is because it has already been turned on at least once, and thus the 'is-new-service' property should be set to false for safety reasons, or if you are performing a timelock migration you should remove this node from the known bootstrapping node list.
```